File: src/main/kotlin/com/gyh/fleacampus/mapper/RoleMapper.java
package com.gyh.fleacampus.mapper;


import com.gyh.fleacampus.model.Role;
import com.gyh.fleacampus.model.UserRole;

import java.util.List;

/**
 * Created by gyh on 2021/2/4
 */
public interface RoleMapper {
    List<Role> findRoleByUserId(Integer userId);

    Integer insert(Integer userId, Integer roleId, Integer unitId);

    List<Role> findAll();

    Integer findIdByName(String name);

    List<UserRole> findByUserIdAndRoleId(Integer userId, Integer roleId);

    Integer removeRoleToUser(Integer userId, Integer roleId, Integer unitId);

    Integer deleteById(Integer id);

    Integer updateRoleById(Integer id, Integer roleId, Integer unitId);

    Integer updateRoleByUserIdAndUnitId(Integer userId, Integer roleId, Integer unitId);
}

File: src/main/kotlin/com/gyh/fleacampus/mapper/RoleMapper.kt
package com.gyh.fleacampus.mapper

import com.gyh.fleacampus.model.Role
import com.gyh.fleacampus.model.UserRole

/**
 * Created by gyh on 2021/2/4
 */
interface RoleMapper {
    fun findRoleByUserId(userId: Int): List<Role>
    fun insert(userId: Int, roleId: Int): Int
    fun findAll(): List<Role>
    fun findIdByName(name: String): Int
    fun findByUserIdAndRoleId(userId: Int, roleId: Int): List<UserRole>
    fun removeRoleToUser(userId: Int, roleId: Int): Int
    fun deleteById(id: Int): Int
    fun updateRoleById(id: Int, roleId: Int): Int
    fun updateRoleByUserIdAndUnitId(userId: Int, roleId: Int): Int
}